normative-conventions.md

```diff
@@ -0,0 +1,25 @@
+# Normative conventions
+
+This document describes some of the conventions that we try to follow when designing new features. Sometimes old features don't follow these; that's ok, new features generally should anyway.
+
+None of these rules are inviolable, but you should have a good reason for any particular feature to deviate (for example, because the feature is a close cousin of an existing feature, like `findLastIndex` vs `findIndex`).
+
+This list is very far from being complete.
+
+## Number-taking inputs should reject `NaN`
+
+In any built-in function which expects a number (including as an option in an options bag), receiving `NaN` or anything which results in `NaN` after coercion is performed (such as by passing through `ToNumeric`, `ToPrimitive`, `ToNumber`, etc.) should cause a `RangeError` to be thrown.
+
+Note that some abstract operations, like [ToIntegerOrInfinity](https://tc39.es/ecma262/multipage/abstract-operations.html#sec-tointegerorinfinity), [ToIndex](https://tc39.es/ecma262/multipage/abstract-operations.html#sec-toindex), etc, will coerce inputs to Number and then further coerce the result, sometimes including mapping `NaN` to non-`NaN` values. So you'll need to coerce inputs to a Number using `ToNumber` before calling those operations, check for `NaN`, and then pass the resulting Number to the operation.
+
+An exception to this rule is functions which take optional numeric arguments: in that case, receiving `undefined` should be treated as the argument not being passed.
+
+NB: This convention is new as of 2023, and most earlier parts of the language do not follow it.
```

Comment:
I think all normative conventions will always have grandfathered-in exceptions, so I'd prefer "established as of YYYY" to be standard frontmatter for all conventions instead of an NB, which suggests it'd be exceptional.

Reply:
I can tweak the wording. The thing I want to get at is that some of the conventions are going to document rules we've usually been following already (e.g. "validate the receiver, then the arguments, in order"), whereas some are a break from the past, like these.